ionoscloud/resource_autoscaling_group_test.go

```diff
@@ -0,0 +1,563 @@
+//go:build all || autoscaling
+// +build all autoscaling
+
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	autoscaling "github.com/ionos-cloud/sdk-go-vm-autoscaling"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+const resourceAGName = constant.AutoscalingGroupResource + "." + constant.AutoscalingGroupTestResource
+
+func TestAccAutoscalingGroupBasic(t *testing.T) {
+	var autoscalingGroup autoscaling.Group
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactories,
+		ExternalProviders: randomProviderVersion343(),
+		CheckDestroy:      testAccCheckAutoscalingGroupDestroyCheck,
+		Steps: []resource.TestStep{
+			{
+				Config: testAG_ConfigBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAutoscalingGroupExists(resourceAGName, &autoscalingGroup),
+					resource.TestCheckResourceAttr(resourceAGName, "name", constant.AutoscalingGroupTestResource),
+					resource.TestCheckResourceAttrPair(resourceAGName, "datacenter_id", constant.DatacenterResource+".autoscaling_datacenter", "id"),
+					resource.TestCheckResourceAttr(resourceAGName, "max_replica_count", "5"),
+					resource.TestCheckResourceAttr(resourceAGName, "min_replica_count", "1"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.metric", "INSTANCE_CPU_UTILIZATION_AVERAGE"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.range", "PT24H"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.amount", "1"),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.amount_type", string(autoscaling.ACTIONAMOUNT_ABSOLUTE)),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.termination_policy_type", string(autoscaling.TERMINATIONPOLICYTYPE_OLDEST_SERVER_FIRST)),
+					resource.TestCheckResourceAttr(resourceAGName, "policy.0.scale_in_action.0.cooldown_period", "PT5M"),
```

Comment:
@cristiGuranIonos the check for `delete_volumes` is missing, this happens for all the tests, most probably the other fields that are missing for this test are also missing for the others, check the comments below.